### ISSUE TITLE
Scaffold-DbContext command keep logging even when -Verbose parameter is not used

### ISSUE THREAD

**Issue by indrajitjadeja** [dotnet/efcore#30770]
When I perform EF core Scaffolding/Reverse Engineering operation without using the `-Verbose` parameter.

```
Scaffold-DbContext "ConnectionString" Microsoft.EntityFrameworkCore.SqlServer -OutputDir Model1
```

I have noticed that methods of the `SqlServerLoggerExtensions.cs` class use the `IDiagnosticsLogger<DbLoggerCategory.Scaffolding>.ShouldLog()` method, which returns true and calls the below method

```C#
definition.Log();
````

For Example: line no [257](https://github.com/dotnet/efcore/blob/main/src/EFCore.SqlServer/Extensions/Internal/SqlServerLoggerExtensions.cs#L257)

The `ShouldLog` method should return false whenever the `-Verbose` parameter is not used by the application user.

As methods of the `SqlServerLoggerExtensions.cs` class are getting called on each iteration, when there will be a huge number of tables, columns, indexes, and keys getting processed, writing logs when the `-Verbose` parameter is not used will impact the execution time of the Scaffold-DbContext command.

EF Core version: 7.0.4
Database provider: Microsoft.EntityFrameworkCore.SqlServer
Target framework: .NET 7.0
Operating system: Windows 11
IDE: Visual Studio 2022 17.5.2

**Comment by indrajitjadeja** [comment]
@ajcvickers, please look into this bug

**Comment by ajcvickers** [comment]
Duplicate of #20989, which is dependent on #8350.

**Comment by indrajitjadeja** [comment]
@ajcvickers When will bug #8350 get fixed?
